File: lib/data/model/movie_response.dart
import 'package:dummy_project/data/model/movie_model.dart';

class MovieResponse {
  final int page;
  final List<MovieModel> results;
  final int totalPages;
  final int totalResults;

  MovieResponse({
    required this.page,
    required this.results,
    required this.totalPages,
    required this.totalResults,
  });

  factory MovieResponse.fromJson(Map<String, dynamic> json) {
    return MovieResponse(
        page: json['page'],
        results: (json['results'] as List)
            .map((e) => MovieModel.fromNetwork(e))
            .toList(),
        totalPages: json['total_pages'],
        totalResults: json['total_results']);
  }
}

File: android/app/src/main/kotlin/com/gayuh/movies/core/remote/dto/MovieResponse.kt
package com.gayuh.movies.core.remote.dto

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class MovieResponse(
    @Json(name = "page")
    val page: Int = 1,

    @Json(name = "results")
    val results: List<MovieModel> = emptyList(),

    @Json(name = "total_pages")
    val totalPages: Int = 0,

    @Json(name = "total_results")
    val totalResults: Int = 0
)
